examples/nodejs/get-set-batch-perf-test/perf-test.ts

```diff
@@ -0,0 +1,239 @@
+    momento: CacheClient,
+    context: PerfTestContext,
+    setConfig: GetSetConfig
+  ): Promise<void> {
+    const setPromises: Promise<CacheSet.Response>[] = [];
+    for (let i = 0; i < setConfig.batchSize; i++) {
+      const key = `key-${i}`;
+      const value = 'x'.repeat(setConfig.itemSizeBytes);
+      const setPromise = this.asyncSet(momento, context, key, value);
+      setPromises.push(setPromise);
+      context.totalItemSizeBytes += setConfig.itemSizeBytes;
+    }
+    await Promise.all(setPromises);
+  }
+
+  private async sendAsyncGetRequests(
+    momento: CacheClient,
+    context: PerfTestContext,
+    getConfig: GetSetConfig
+  ): Promise<void> {
+    const getPromises: Promise<CacheSet.Response>[] = [];
+    for (let i = 0; i < getConfig.batchSize; i++) {
+      const key = `key-${i}`;
+      const getPromise = this.asyncGet(momento, context, key);
+      getPromises.push(getPromise);
+      context.totalItemSizeBytes += getConfig.itemSizeBytes;
+    }
+    await Promise.all(getPromises);
+  }
+
+  private async sendSetBatchRequests(
+    momento: CacheClient,
+    context: PerfTestContext,
+    setConfig: GetSetConfig
+  ): Promise<void> {
+    const keys = Array.from({length: setConfig.batchSize}, (_, i) => `key-${i}`);
+    const values = Array.from({length: setConfig.batchSize}, () => 'x'.repeat(setConfig.itemSizeBytes));
+    const items = new Map<string, string>();
+    keys.forEach((key, index) => {
+      items.set(key, values[index]);
+    });
+    const setBatchStartTime = process.hrtime();
+    const setBatchPromise = momento.setBatch(this.cacheName, items);
+
+    void setBatchPromise.then(() => {
+      const setBatchDuration = getElapsedMillis(setBatchStartTime);
+      context.setBatchLatencies.recordValue(setBatchDuration);
+    });
+    context.totalItemSizeBytes += setConfig.batchSize * setConfig.itemSizeBytes;
+    await setBatchPromise;
+  }
+
+  private async sendGetBatchRequests(
+    momento: CacheClient,
+    context: PerfTestContext,
+    getConfig: GetSetConfig
+  ): Promise<void> {
+    const keys = Array.from({length: getConfig.batchSize}, (_, i) => `key-${i}`);
+    const getBatchStartTime = process.hrtime();
+    const getBatchPromise = momento.getBatch(this.cacheName, keys);
+    void getBatchPromise.then(() => {
+      const getBatchDuration = getElapsedMillis(getBatchStartTime);
+      context.getBatchLatencies.recordValue(getBatchDuration);
+    });
+    context.totalItemSizeBytes += getConfig.batchSize * getConfig.itemSizeBytes;
+    await getBatchPromise;
+  }
+
+  private asyncSet(
+    momento: CacheClient,
+    context: PerfTestContext,
+    key: string,
+    value: string
+  ): Promise<CacheSet.Response> {
+    const startTime = process.hrtime();
+    const setPromise = momento.set(this.cacheName, key, value);
+    void setPromise.then(() => {
+      const setDuration = getElapsedMillis(startTime);
+      context.asyncSetLatencies.recordValue(setDuration);
```

Comment:
I think here you are measuring the latency for each individual set request. So your `count` value in your histogram is going to be much much higher than the `count` value for the number of calls to `setBatch`.

What we really want to compare is the time it takes to do one `setBatch` vs. the time it takes to do **a full batch** of calls to `set`. So I think you will want to push this call to `recordValue` up into `sendAsyncSetRequests`. And the same for `get`.

A good sanity check might be to see what the `count` in the histograms look like at the end of each run. For a given configuration (batch size + item size), I would expect the `count` in the histogram to be relatively close to the same value for both `set` and `setBatch`. (I would expect that with your current code, with a batch size of 10, the `count` would differ by approximately a factor of 10; with a batch size of 100, it would differ by approximately a factor of `100`, etc.)

Reply:
I was taking care of the count for batch apis when calculating the summary. But, you're right I am still comparing latencies for one set request vs batch of set requests.